### ISSUE TITLE
Libraries.io consumer is creating junk PyPi projects

### ISSUE THREAD

**Issue by yarons** [fedora-infra/anitya#881]
I'm not sure what happened but it seems like many new PyPi project were added, looking at all the projects page I'm seeing that there are many projects with the same version number and links that doesn't work.

What's up? Maybe it's a malicious act that requires a rollback?

**Comment by Zlopez** [comment]
This could be caused by libreries.io consumer, which is listening to every project from PyPi and Rubygems. But this should be solved by https://github.com/release-monitoring/anitya/issues/829 in the future.

**Comment by yarons** [comment]
But, what are we doing to remove the entries now? Will this fix remove these entries from the current DB?

**Comment by Zlopez** [comment]
Yes, it will remove every project, that couldn't be checked for new version after few days.

**Comment by yarons** [comment]
Hmmm but what about projects with a temporary website issue?

**Comment by Zlopez** [comment]
If the project ever retrieved any version it will not be deleted. This automation will only delete projects without any version, so only the ones that have incorrect setup.

**Comment by yarons** [comment]
Cool, sounds better.

**Comment by Zlopez** [comment]
This will be part of next release, which is now waiting for last PR to be merged. So hopefully this will be out soon.

**Comment by Zlopez** [comment]
The projects created by libraries.io consumer had at least one version. So this new automatic will unfortunately not remove these junk projects. I need to look at the consumer and see why is this happening.

**Comment by Zlopez** [comment]
After looking at the libraries.io consumer script, I can tell that this project must have existed on PyPi for some time and probably was only created and then removed from PyPi. So I will try to clean the projects created by this situation, but this looks like issue on PyPi side and I'm not sure if we actually could prevent this.
I could probably expand the automatic later to also remove the project with only one retrieved version, but this could potentially remove good project.

**Comment by yarons** [comment]
Cool, all I could notice is that the name contsins 2 characters and that the names are increasing by 1 and the only version is 0.0.1.

**Comment by Zlopez** [comment]
I did something better in #882, I started to use mapping to distro as another condition. So with this change it will delete projects, that reach error counter and have no mapping. If they have mapping, but don't have any version, they are still deleted.

This should remove many junk projects, that didn't received any version and only throwing error, or doesn't receive versions anymore and didn't have any mapping to downstream. This should clear Anitya db a little.
